### DISCUSSION TITLE
Deployment using AWS EFS for Dgraph storage?

### DISCUSSION THREAD

**Discussion by pchaganti** [cloudgraphdev/cli#35] [General]
Is it possible to deploy cloudgraphdev to AWS and use EFS for the storage needed by Dgraph? Trying to understand deployment options.

thanks

**Comment by ckoning** [comment]
It would be possible to set up DGraph to do this via a few different methods, like using an EFS persistent volume in EKS or building containers on the DGraph provided ones that configure and mount the EFS volume to the container before starting the service.

However, is not a good idea. E

FS has small but non negligible propagation delays for internal replication. This causes problems for very rapid write-read workloads. It is advised not to use EFS for the data storage for PostgreSQL, for example, as it can cause data loss in some usage patterns. I've personally encountered this trying to use EFS as the storage backend for a git repository. We haven't done any profiling, but I expect this would be the case with CloudGraph's usage of DGraph.

**Reply by pchaganti** [reply]
In order to use it on AWS, what would you recommend for storage? EBS volumes?

thanks!
